src/sssom/parsers.py

```diff
@@ -182,6 +182,97 @@ def _get_seperator_symbol_from_file_path(file):
     return None
 
 
+def _is_check_valid_extension_slot(slot_name):
+    logging.warning(
+        f"'{slot_name}' could be a valid extension slot "
+        f"(https://mapping-commons.github.io/sssom/spec-model/#non-standard-slots), "
+        f"but the validator does not check that yet."
+    )
+    return False
+
+
+def _check_irregular_metadata(sssom_metadata, meta):
+    fail_metadata = False
+    for m in [sssom_metadata, meta]:
+        for key in m:
+            if (key not in _get_sssom_schema_object().mapping_set_slots) and (
+                not _is_check_valid_extension_slot(key)
+            ):
+                logging.warning(
+                    f"Metadata key '{key}' is not a standard SSSOM mapping set metadata field."
+                )
+                fail_metadata = True
+    return fail_metadata
+
+
+def _check_redefined_builtin_prefixes(sssom_metadata, meta, prefix_map):
+
+    # There are three ways in which prefixes can be communicated, so we will check all of them
+    # This is a bit overly draconian, as in the end, only the highest priority one gets picked
+    # But since this only constitues a (logging) warning, I think its worth reporting
+    builtin_converter = _get_built_in_prefix_map()
+    sssom_metadata_converter = _get_converter_pop_replace_curie_map(sssom_metadata)
+    meta_converter = _get_converter_pop_replace_curie_map(meta)
+    prefix_map_converter = ensure_converter(prefix_map, use_defaults=False)
+    is_valid_prefixes = True
+
+    for converter in [sssom_metadata_converter, meta_converter, prefix_map_converter]:
+        for builtin_prefix, builtin_uri in builtin_converter.bimap.items():
+            if builtin_prefix in converter.bimap:
+                if builtin_uri != converter.bimap[builtin_prefix]:
+                    logging.warning(
+                        f"A built-in prefix ({builtin_prefix}) was provided, "
+                        f"but the provided URI expansion ({converter.bimap[builtin_prefix]}) does not correspond "
+                        f"to the required URI expansion: {builtin_uri}. The prefix will be ignored."
+                    )
+                    is_valid_prefixes = False
+            reverse_bimap = {value: key for key, value in builtin_converter.bimap.items()}
+            if builtin_uri in reverse_bimap:
+                if builtin_prefix != reverse_bimap[builtin_uri]:
+                    logging.warning(
+                        f"A built-in URI namespace ({builtin_uri}) was used in (one of) the provided prefix map(s), "
+                        f"but the provided prefix ({reverse_bimap[builtin_uri]}) does not correspond to the "
+                        f"standard prefix: {builtin_prefix}. The prefix will be ignored."
```

Comment:
I’ll just note that this case (standard prefix associated to a different prefix name than the one listed in the spec, e.g. `som: https://w3id.org/sssom/`) is _not_ forbidden by the spec. The spec only forbids associated a builtin prefix name to a different prefix (e.g., `sssom: https://not-the-sssom-namespace.example.org/`).

The spec says:

> By exception, prefix names listed in the table found in the IRI prefixes section are considered “built-in”. As such, they MAY be omitted from the curie_map. If they are not omitted, they MUST point to the same IRI prefixes as in the aforementioned table.

There is nothing here that forbids using another prefix name than a builtin one to refer to one of the known namespaces.

Not requesting any change here because SSSOM-Py is free to be stricter than the spec if it wants to.